### PR TITLE
Add general setting to configure session timeout

### PR DESCRIPTION
**Pull request by NoItemName**
Enables session timeout configuration. This feature doesn't override _remember me_ checkbox.
Implements [#619](https://github.com/airsonic-advanced/airsonic-advanced/issues/619)

### CODE REVIEW

**Comment by randomnicode** [comment]
Does this spring config not suffice?
`server.servlet.session.timeout=60s`
specified in https://docs.spring.io/spring-boot/docs/current/reference/html/application-properties.html#application-properties.server.server.servlet.session.timeout

It's a good idea to just use the spring property without overriding manually in a custom config class. okay to have a ui for specifying it. We will provide PR comments accordingly

File: airsonic-main/src/main/java/org/airsonic/player/service/SettingsService.java [hunk 1]
```diff
@@ -86,6 +86,7 @@
     private static final String KEY_WELCOME_SUBTITLE = "WelcomeSubtitle";
     private static final String KEY_WELCOME_MESSAGE = "WelcomeMessage2";
     private static final String KEY_LOGIN_MESSAGE = "LoginMessage";
+    private static final String KEY_SESSION_TIMEOUT_MESSAGE = "SessionTimeout";
```
Comment (randomnicode) [inline airsonic-main/src/main/java/org/airsonic/player/service/SettingsService.java:89]: Probably should be named `server.servlet.session.timeout` so it overrides the spring boot value

File: airsonic-main/src/main/java/org/airsonic/player/service/SettingsService.java [hunk 2]
```diff
@@ -188,6 +189,7 @@
             "\\\\ \\\\\n" +
             "To change or remove this message, log in with administrator rights and go to <a href='settings.view'>Settings</a> > <a href='generalSettings.view'>General</a>.";
     private static final String DEFAULT_LOGIN_MESSAGE = null;
+    private static final int DEFAULT_SESSION_TIMEOUT_MINUTES = 60;
```
Comment (randomnicode) [inline airsonic-main/src/main/java/org/airsonic/player/service/SettingsService.java:192]: Maybe a String or Duration?

org.springframework.boot.web.servlet.server.Session details the timeout field as a Duration

default is 30m it seems

File: airsonic-main/src/main/java/org/airsonic/player/spring/CustomSessionListener.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+import org.airsonic.player.service.SettingsService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Configuration;
+
+import javax.servlet.http.HttpSessionEvent;
+import javax.servlet.http.HttpSessionListener;
+
+@Configuration
+public class CustomSessionListener implements HttpSessionListener {
```
Comment (randomnicode) [inline airsonic-main/src/main/java/org/airsonic/player/spring/CustomSessionListener.java:11]: this class isn't needed

File: airsonic-main/src/main/resources/org/airsonic/player/i18n/ResourceBundle_en.properties [hunk 4]
```diff
@@ -373,6 +373,7 @@ generalsettings.welcometitle=Welcome title
 generalsettings.welcomesubtitle=Welcome subtitle
 generalsettings.welcomemessage=Welcome message
 generalsettings.loginmessage=Login message
+generalsettings.sessiontimeout=Session timeout in minutes. Changes will take effect on a new login
```
Comment (randomnicode) [inline airsonic-main/src/main/resources/org/airsonic/player/i18n/ResourceBundle_en.properties:376]: if it's a string, session timeout will be a duration string, if no postfix then in seconds.
also not sure when changes take effect with the spring boot var, maybe at application restart?

File: airsonic-main/src/main/java/org/airsonic/player/command/GeneralSettingsCommand.java [hunk 5]
```diff
@@ -172,6 +173,14 @@ public void setLoginMessage(String loginMessage) {
         this.loginMessage = loginMessage;
     }
 
+    public Integer getSessionTimeout() {
```
Comment (randomnicode) [inline airsonic-main/src/main/java/org/airsonic/player/command/GeneralSettingsCommand.java:176]: string or duration?

**Review by randomnicode — CHANGES_REQUESTED** [review]

**Comment by NoItemName** [comment]
Thanks for the feedback, i will work on the changes!

**Review by randomnicode — APPROVED** [review]

**Pull request merged**